src/geom/TileMatrixSet.js

```diff
@@ -0,0 +1,143 @@
+
+        /**
+         * Create a TileMatrixSet based on a quad division technique given the provided initial starting conditions.
+         * @param sector the geographic bounding area of the TileMatrixSet
+         * @param matrixWidth the number of tiles in the x direction at the initial level
+         * @param matrixHeight the number of tiles in the y direction at the initial level
+         * @param tileWidth the number of pixels or points in the x direction of a tile
+         * @param tileHeight the number of pixels or points in the y direction of a tile
+         * @param numLevels the number of resolution levels this TileMatrixSet should contain
+         * @returns {TileMatrixSet} fully configured TileMatrixSet
+         */
+        TileMatrixSet.fromTilePyramid = function (sector, matrixWidth, matrixHeight, tileWidth, tileHeight, numLevels) {
+            if (!sector) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrixSet", "fromTilePyramid", "missingSector"));
+            }
+
+            if (matrixWidth < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrixSet", "fromTilePyramid", "invalidWidth"));
+            }
+
+            if (matrixHeight < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrixSet", "fromTilePyramid", "invalidHeight"));
+            }
+
+            if (tileWidth < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrixSet", "fromTilePyramid", "invalidWidth"));
+            }
+
+            if (tileHeight < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrixSet", "fromTilePyramid", "invalidHeight"));
+            }
+
+            if (numLevels < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrixSet", "fromTilePyramid",
+                        "The specified number of levels is invalid"));
+            }
+
+            var tileMatrices = [], matrix, idx;
+
+            for (idx = 0; idx < numLevels; idx++) {
+                matrix = new TileMatrix(sector, idx, matrixWidth, matrixHeight, tileWidth, tileHeight);
+                tileMatrices.push(matrix);
+                matrixWidth *= 2;
+                matrixHeight *= 2;
+            }
+
+            return new TileMatrixSet(sector, tileMatrices);
+        };
+
+        /**
+         * Determines the index of the TileMatrix with the closest resolution to the provided value.
+         * @param degreesPerPixel the target resolution
+         * @returns {number} the index of the TileMatrix within this TileMatrixSet's entries array
+         */
+        TileMatrixSet.prototype.indexOfMatrixNearest = function (degreesPerPixel) {
+            if (degreesPerPixel < 0) {
```

Comment:
This check could be '<=0', could it not?

Reply:
@markpet49 On second thought, is there a value in allowing zero, say for requesting the highest resolution case?

Reply:
For my edification, what about when `0` is a legal value? What should a check look like?

Reply:
Nevertheless, @markpet49 highlights an important point. The argument checking pattern requires a judgement call, and different authors have been making different decisions.

As Mark suggested, how about we use the standard check when zero is not a legal value:
`if (!degreesPerPixel)`